Integration tests opt2

MockerTim opened the pull request:
Hi Sander,
I've moved the integration tests to the distribution module, and made the custom naming of the test class possible.
I had to make additional module just to make JUnit custom Category available to all modules. 
I also had to make some tricky configurations to all modules. 
So, I decided to add this changes to the separate pull request.

sfieten commented:
Hi Timur,
I think that the extra module might actually be even better especially when we can also move all integration test code into this module. That way all modules have their own "target". 
Could therefore check if all integration test code can be moved into one _Integration Test_ module?

MockerTim commented:
Hi Sander,
I've moved integration tests to the separate module. 

I should have done this from the beginning. My fault. The professional tester should be wiser.

I need to make a comment. I've moved integration tests themself to the holodeckb2b-it module and added another holodeckb2b-it_lib module which contains only the IntegrationTest junit category definition. Although the holodeckb2b-it module is the only one that contains integration tests now, the holodeckb2b-it_lib  module is still needed for testing, because the main pom uses the IntegrationTest category during the building of all modules that contain junit tests.

sfieten commented:
Hi Timur,
I prefer to have just one module related to integration testing. Do you think and can you check if it is possible to use the failsafe plugin only in the integration test module or have only the specific integration test config there?
If that isn't possible I prefer to move the interface that defines the integration test category to the Common module. And if that is not possible we should just abandon the idea of a custom defined category.

MockerTim commented:
Hi Sander,

I think it is possible, but we'll need to move the
org.holodeckb2b.integ.IntegrationTest interface
to the holodeckb2b-interfaces module src/main/java folder.
In such a way holodeckb2b-interfaces module will contain interface
definition
that does not correspond to it's logic but is used only for integration
testing.
Do you think this variant will be acceptable?

2017-02-08 16:26 GMT+03:00 Sander Fieten <notifications@github.com>:

> Hi Timur,
> I prefer to have just one module related to integration testing. Do you
> think and can you check if it is possible to use the failsafe plugin only
> in the integration test module or have only the specific integration test
> config there?
> If that isn't possible I prefer to move the interface that defines the
> integration test category to the Common module. And if that is not possible
> we should just abandon the idea of a custom defined category.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/holodeck-b2b/Holodeck-B2B/pull/42#issuecomment-278328078>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA1cCZ4ERqeUV2yhFYGZsaIWkN6XLykIks5racKVgaJpZM4L3gX6>
> .
>



-- 
С уважением,
Шакуов Тимур

sfieten commented:
Hi Timur,

as the interfaces module is more on the public interfaces and the integration test definition is only needed internally I think the common module would be a better place for it. But if that really isn’t possible we should maybe abandon the idea of creating the category and just use the standard failsafe configuration. The category is also less relevant now we moved integration testing to its own module. Just give it some thoughts, but don’t hesitate to remove the category again. We learned how it works but also found out it is not useful for the project (right now).


> On 8 Feb 2017, at 14:59, Timur Shakuov <notifications@github.com> wrote:
> 
> Hi Sander,
> 
> I think it is possible, but we'll need to move the
> org.holodeckb2b.integ.IntegrationTest interface
> to the holodeckb2b-interfaces module src/main/java folder.
> In such a way holodeckb2b-interfaces module will contain interface
> definition
> that does not correspond to it's logic but is used only for integration
> testing.
> Do you think this variant will be acceptable?
> 
> 2017-02-08 16:26 GMT+03:00 Sander Fieten <notifications@github.com>:
> 
> > Hi Timur,
> > I prefer to have just one module related to integration testing. Do you
> > think and can you check if it is possible to use the failsafe plugin only
> > in the integration test module or have only the specific integration test
> > config there?
> > If that isn't possible I prefer to move the interface that defines the
> > integration test category to the Common module. And if that is not possible
> > we should just abandon the idea of a custom defined category.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/holodeck-b2b/Holodeck-B2B/pull/42#issuecomment-278328078>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AA1cCZ4ERqeUV2yhFYGZsaIWkN6XLykIks5racKVgaJpZM4L3gX6>
> > .
> >
> 
> 
> 
> -- 
> С уважением,
> Шакуов Тимур
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/holodeck-b2b/Holodeck-B2B/pull/42#issuecomment-278335910>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AJC3I8I3Wmi88VXgC8ZbR1ira0EpiiEAks5racpZgaJpZM4L3gX6>.
>

MockerTim commented:
Hi Sander,
I moved the integration tests dependencies to the common module.
There is only one module for integration testing now.

sfieten commented:
Hi Timur,
I did some experimentation on the code from your pull request to try to move everything related to integration testing to the integration test module. After some changes to the pom files of the main project and the integration test module I managed to get everything in one module.
Please see my version at https://github.com/sfieten/Holodeck-B2B/tree/integration_tests_opt2
Can you please check this code and make sure it also works in your development environment? If it does and you are okay with this change can you integrate it into your code? 
I was also thinking if the package names as currently used are the most logical and whether `org.holodeckb2b.test.integration` would not be better?

MockerTim commented:
Hi Sander,
Cool! Checked your modifications. The code works fine.
Merged your changes.
Sure. The `org.holodeckb2b.test.integration` sounds better for integration tests package.
I've renamed `org.holodeckb2b.integ`.
